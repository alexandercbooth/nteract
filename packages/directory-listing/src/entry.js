// @flow
import * as React from "react";

import { areComponentsEqual } from "react-hot-loader";

import TimeAgo from "@nteract/timeago";
import { Book, FileText, FileDirectory } from "@nteract/octicons";

type EntryProps = { children?: React.Node };

export default class Entry extends React.Component<EntryProps, *> {
  static Icon = ({
    fileType
  }: {
    fileType: "unknown" | "notebook" | "directory" | "file" | "dummy"
  }) => {
    let icon = <FileText />;
    switch (fileType) {
      case "notebook":
        icon = <Book />;
        break;
      case "directory":
        icon = <FileDirectory />;
        break;
      case "file":
        icon = <FileText />;
        break;
      default:
        icon = <FileText />;
    }

    return (
      <td className="icon">
        {icon}
        <style jsx>{`
          :global(.icon) {
            padding-right: 2px;
            padding-left: 10px;
            width: 17px;
            vertical-align: middle;
            text-align: center;
            opacity: 0.95;
            color: #0366d6;
          }
        `}</style>
      </td>
    );
  };

  static Name = ({ link }: { link: any }) => (
    <td className="name">
      {link}
      <style jsx>{`
        .name {
          vertical-align: middle;
          font-size: 0.9em;
          padding: 8px;
        }

        :global(a) {
          text-decoration: none;
        }

        :global(a:hover) {
          text-decoration: underline;
          outline-width: 0;
        }
      `}</style>
    </td>
  );
  static LastSaved = ({ last_modified }: { last_modified: Date }) => (
    <td className="timeago">
      <TimeAgo date={last_modified} />
      <style jsx>{`
        .timeago {
          text-align: right;
          color: #6a737d;
          padding-right: 10px;
        }
      `}</style>
    </td>
  );
  render() {
    return (
      <tr className="directory-entry">
        {React.Children.map(this.props.children, child => {
          if (
            areComponentsEqual(child.type, Entry.Icon) ||
            areComponentsEqual(child.type, Entry.Name) ||
            areComponentsEqual(child.type, Entry.LastSaved)
          ) {
            return React.cloneElement(child, {
              className:
                typeof child.props.className === String &&
                child.props.className !== ""
                  ? child.props.className + " directory-entry-field"
                  : "directory-entry-field"
            });
          } else {
            return <td className="directory-entry-field">{child}</td>;
          }
        })}
        <style jsx>{`
          tr {
            border-top: 1px solid #eaecef;
          }

          tr:hover {
            background-color: #f6f8fa;
            transition: background-color 0.1s ease-out;
          }

          tr:first-child {
            border-top: none;
          }

          tr:last-child {
            border-bottom: none;
          }
        `}</style>
      </tr>
    );
  }
}
