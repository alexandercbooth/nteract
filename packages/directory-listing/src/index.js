// @flow
import Entry from "./entry";
import Listing from "./listing";
export { Entry };
export default Listing;
