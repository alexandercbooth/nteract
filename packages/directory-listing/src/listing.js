// @flow
import * as React from "react";

export type ListingProps = { children: React.Node };

export default class Listing extends React.Component<ListingProps> {
  render() {
    return (
      <React.Fragment>
        <div>
          <table>
            <tbody>{this.props.children}</tbody>
          </table>
        </div>
        <style jsx>{`
          tr {
            border-top: 1px solid #eaecef;
          }

          td a {
            text-decoration: none;
          }

          td a:hover {
            text-decoration: underline;
            outline-width: 0;
          }

          tr:hover {
            background-color: #f6f8fa;
            transition: background-color 0.1s ease-out;
          }

          tr:first-child {
            border-top: none;
          }

          tr:last-child {
            border-bottom: none;
          }

          table {
            width: 100%;
            border-collapse: collapse;
            border-radius: 2px;
            border-spacing: 0;
          }

          td a:hover {
            outline-width: 0;
            text-decoration: underline;
          }
        `}</style>
      </React.Fragment>
    );
  }
}
