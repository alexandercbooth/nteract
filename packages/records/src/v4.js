// @flow

/*
 * Functions in this module are provided for converting from Jupyter Notebook
 * Format v4 to nteract's in-memory format, affectionately referred to as
 * commutable.
 *
 * See: https://github.com/jupyter/nbformat/blob/62d6eb8803616d198eaa2024604d1fe923f2a7b3/nbformat/v4/nbformat.v4.schema.json
 *
 * The main goal here is consistency and compliance with the v4 spec. The types
 * contained in here (non Immutable ones) are constrained to the disk based
 * notebook format.
 *
 * To assist in the developer experience, types are included through the use of
 * flow.
 *
 */

import type { NbformatCell } from "./cells";

import type { NbformatOutput } from "./outputs";

import type {
  ImmutableNotebook,
  ImmutableCodeCell,
  ImmutableMarkdownCell,
  ImmutableRawCell,
  ImmutableCell,
  ImmutableOutput,
  ImmutableMimeBundle
} from "./types";

import type { CellStructure } from "./structures";

const appendCell = require("./structures").appendCell;
