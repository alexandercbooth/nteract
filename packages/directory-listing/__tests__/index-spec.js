// @flow
import * as React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { Entry } from "../src";
import Listing from "../src";

describe("Listing", () => {
  it("accepts props and renders entries in directory in a table", () => {
    const component = (
      <Listing>
        <Entry>
          <Entry.Icon fileType={"directory"} />
          <Entry.Name link={"linky"} />
          <Entry.LastSaved
            last_modified={
              "Fri Jun 22 2018 00:15:55 GMT-0400 (Eastern Daylight Time)"
            }
          />
        </Entry>
      </Listing>
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});

describe("Entry", () => {
  it("accepts props and renders entries in directory", () => {
    const component = shallow(
      <Entry>
        <Entry.Icon fileType={"directory"} />
        <Entry.Name link={"linky jr"} />
        <Entry.LastSaved
          last_modified={
            "Fri Jun 22 2018 00:15:55 GMT-0400 (Eastern Daylight Time)"
          }
        />
      </Entry>
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});
