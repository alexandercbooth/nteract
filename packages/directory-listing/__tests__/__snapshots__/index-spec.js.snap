// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entry accepts props and renders entries in directory 1`] = `
<tr
  className="directory-entry"
>
  <Component
    className="directory-entry-field"
    fileType="directory"
    key=".0"
  />
  <Component
    className="directory-entry-field"
    key=".1"
    link="linky jr"
  />
  <Component
    className="directory-entry-field"
    key=".2"
    last_modified="Fri Jun 22 2018 00:15:55 GMT-0400 (Eastern Daylight Time)"
  />
</tr>
`;

exports[`Listing accepts props and renders entries in directory in a table 1`] = `undefined`;
